devlib/module/sched.py

```diff
@@ -0,0 +1,239 @@
+        cpus = range(0, self.target.number_of_cpus)
+
+        capacities = {cpu : 0 for cpu in cpus}
+
+        for cpu in cpus:
+            if self._has_em(cpu):
+                capacities[cpu] = self._get_em_capacity(cpu)
+            elif self._has_dmips_capacity(cpu):
+                capacities[cpu] = self._get_dmips_capacity(cpu)
+            else:
+                self.logger.warning('Cannot read capacity for cpu{}, assuming 1024'.format(cpu))
+                capacities[cpu] = 1024
+
+        return capacities
+
+    @memoized
+    def get_hz(self):
+        """
+        :returns: The scheduler tick frequency on the target
+        """
+        return int(test_env.target.config.get('CONFIG_HZ'))
+
+class SchedDomainData(object):
+    """
+    An interface to the sched domain data exposed in the procfs
+
+    The raw procfs nodes are available under the 'procfs' attribute.
+    They are also parsed and available as attributes of the object.
+
+    -----
+    Example:
+    data = target.sched.get_sd_info(0)
+
+    print info.domains[0].name
+    > 'MC'
+    print info.procfs["domain0"]["name"]
+    > 'MC'
+    """
+
+    sched_domain_root = '/proc/sys/kernel/sched_domain'
+
+    def _build_domains(self):
+        self.domains = {}
+
+        for domain, fields in self.procfs.iteritems():
+            self.domains[ProcFSNode._node_digits(domain)] = SchedDomain(fields)
+
+    def __init__(self, target, cpu):
+        self.procfs = target.read_tree_values(
+            os.path.join(self.sched_domain_root, 'cpu{}'.format(cpu)),
+            depth=5
+        )
+
+        self._build_domains()
+
+class ProcFSNode(object):
+    re_node = re.compile(r"(?P<name>.*)(?P<digits>\d+)$")
+
+    @staticmethod
+    def _ends_with_digits(node):
+        if not isinstance(node, str):
+            return False
+
+        return re.search(ProcFSNode.re_node, node) != None
+
+    @staticmethod
+    def _node_digits(node):
+        """
+        :returns: The ending digits of the procfs node
+        """
+        return int(re.search(ProcFSNode.re_node, node).group("digits"))
+
+    @staticmethod
+    def _node_name(node):
+        """
+        :returns: The name of the procfs node
+        """
+        return re.search(ProcFSNode.re_node, node).group("name")
+
+    @staticmethod
+    def _packable(node, entries):
+        """
+        :returns: Whether it makes sense to pack a node into a common entry
+        """
+        return (ProcFSNode._ends_with_digits(node) and
+                any([ProcFSNode._ends_with_digits(x) and
+                     ProcFSNode._node_digits(x) != ProcFSNode._node_digits(node) and
+                     ProcFSNode._node_name(x) == ProcFSNode._node_name(node)
+                     for x in entries]))
+
+    def _build_node(self, node):
+        if isinstance(node, dict):
+            return ProcFSNode(node)
+        else:
+            value = node
+            try:
+                value = int(node)
+            except:
```

Comment:
Catch-all except are evil and must not be used apart in something like nosetest. That will even catch stuff like SyntaxError, which can be extremely confusing:

```
except ValueError:
```